### PR TITLE
Fix external PolyClipper and add dev build feature

### PR DESCRIPTION
**Pull request by ldowen**
# Summary

- This PR is a bugfix and feature PR
- It does the following:
  - Fixes issue with using external PolyClipper
  - Fixes issue with using newer Axom version than Spack specifies
  - Updates PolyClipper and BLT versions
  - Adds options for using individual C++ shared libraries to reduce build time during development
  - Updated docs with relevant CMake variables

------
### ToDo :

- [x] Annotate ``RELEASE_NOTES.md`` with notable changes.
- [x] Create LLNLSpheral PR pointing at this branch. (PR#63)
- [x] LLNLSpheral PR has passed all tests.

### CODE REVIEW

**Review by jmikeowen — APPROVED** [review]
These changes all look good.  Once the CI tests pass should be good to merge.

File: cmake/spheral/SpheralAddLibs.cmake [hunk 1]
```diff
@@ -77,20 +89,20 @@ function(spheral_add_cxx_library package_name _cxx_obj_list)
   # For including files in submodules, currently unused
   get_property(SPHERAL_SUBMOD_INCLUDES GLOBAL PROPERTY SPHERAL_SUBMOD_INCLUDES)
 
-  if(NOT ENABLE_SHARED)
-    # Build static spheral C++ library
+  if(ENABLE_SHARED)
+    # Build shared spheral C++ library, currently unavailable
```
Comment (mdavis36) [inline cmake/spheral/SpheralAddLibs.cmake:93]: Why is this currently unavailable?

File: src/CXXTests/CMakeLists.txt [hunk 2]
```diff
@@ -1,8 +1,7 @@
-get_property(SPHERAL_CXX_LIBS GLOBAL PROPERTY SPHERAL_CXX_LIBS)
 
 add_subdirectory(DeviceTestLib)
 
-if(ENABLE_TESTS)
+if(ENABLE_TESTS AND NOT ENABLE_DEV_BUILD)
   blt_add_executable( NAME spheral_cuda_test
                       SOURCES Spheral_CUDA_Test.cc
                       DEPENDS_ON Spheral_CXX ${spheral_blt_cxx_depends})
```
Comment (mdavis36) [inline src/CXXTests/CMakeLists.txt:7]: Can we make this work with DEV_BUILD? We should be able to get the list of C++ libraries and make them a dependency right?

File: cmake/spheral/SpheralAddLibs.cmake (hunk 1 above)
Comment (ldowen) [reply cmake/spheral/SpheralAddLibs.cmake:93]: That comment is old. I will remove it.

File: src/CXXTests/CMakeLists.txt (hunk 2 above)
Comment (ldowen) [reply src/CXXTests/CMakeLists.txt:7]: We might be able to reproduce what we do https://github.com/LLNL/spheral/blob/b4b6ec5a5795d5132bfb774e0d16876483473180/cmake/spheral/SpheralAddLibs.cmake#L212

**Review by mdavis36 — APPROVED** [review]

**Pull request merged**